From 954d9fdb962040876158ddcb4bd47482cbdfb13f Mon Sep 17 00:00:00 2001
From: Dylan Yip <dylan.yip@xilinx.com>
Date: Thu, 22 Aug 2019 15:23:59 -0700
Subject: [PATCH 0658/1852] media: v4l: Group Xilinx downstream formats

commit 4a056f64c5c117871d144094c317a4ff5bc34579 from
https://github.com/Xilinx/linux-xlnx.git

This patch groups Xilinx downstream formats together at new offsets
(0x1100 for RGB and 0x2100 for YUV) to help avoid future conflicts.

This patch assigns a new enumeration value to VYYUYY8_1X24 to ensure
that all media bus formats have distinct values.

Signed-off-by: Dylan Yip <dylan.yip@xilinx.com>
Reviewed-by: Hyun Kwon <hyun.kwon@xilinx.com>
Reviewed-by: Satish Kumar Nagireddy <satish.nagireddy.nagireddy@xilinx.com>
State: pending
Signed-off-by: Yaliang Wang <Yaliang.Wang@windriver.com>
---
 .../media/uapi/v4l/subdev-formats.rst         | 24 +++++++-------
 include/uapi/linux/media-bus-format.h         | 32 +++++++++++--------
 2 files changed, 30 insertions(+), 26 deletions(-)

diff --git a/Documentation/media/uapi/v4l/subdev-formats.rst b/Documentation/media/uapi/v4l/subdev-formats.rst
index 162cd5c1580f..0f3bdf437f1a 100644
--- a/Documentation/media/uapi/v4l/subdev-formats.rst
+++ b/Documentation/media/uapi/v4l/subdev-formats.rst
@@ -1526,7 +1526,7 @@ The following tables list existing packed RGB formats.
     * .. _MEDIA-BUS-FMT-RBG101010-1X30:
 
       - MEDIA_BUS_FMT_RBG101010_1X30
-      - 0x101b
+      - 0x1100
       -
       - 0
       - 0
@@ -1670,7 +1670,7 @@ The following table list existing packed 36bit wide RGB formats.
     * .. _MEDIA-BUS-FMT-RBG121212-1X36:
 
       - MEDIA_BUS_FMT_RBG121212_1X36
-      - 0x101c
+      - 0x1101
       -
       - r\ :sub:`11`
       - r\ :sub:`10`
@@ -1880,7 +1880,7 @@ The following table list existing packed 48bit wide RGB formats.
     * .. _MEDIA-BUS-FMT-RBG161616-1X48:
 
       - MEDIA_BUS_FMT_RBG161616_1X48
-      - 0x101d
+      - 0x1102
       -
       -
       -
@@ -5332,7 +5332,7 @@ the following codes.
     * .. _MEDIA-BUS-FMT-VYYUYY10_4X20:
 
       - MEDIA_BUS_FMT_VYYUYY10_4X20
-      - 0x2031
+      - 0x2101
       -
       -
       -
@@ -7231,7 +7231,7 @@ the following codes.
     * .. _MEDIA-BUS-FMT-UYYVYY12-4X24:
 
       - MEDIA_BUS_FMT_UYYVYY12_4X24
-      - 0x202d
+      - 0x2103
       -
       -
       -
@@ -7373,7 +7373,7 @@ the following codes.
     * .. _MEDIA-BUS-FMT-VYYUYY8-1X24:
 
       - MEDIA_BUS_FMT_VYYUYY8_1X24
-      - 0x202c
+      - 0x2100
       -
       -
       -
@@ -7447,7 +7447,7 @@ the following codes.
     * .. _MEDIA-BUS-FMT-VUY10-1X30:
 
       - MEDIA_BUS_FMT_VUY10_1X30
-      - 0x2032
+      - 0x2102
       -
       -
       -
@@ -7556,7 +7556,7 @@ the following codes.
     * .. _MEDIA-BUS-FMT-Y16-1X16:
 
       - MEDIA_BUS_FMT_Y16_1X16
-      - 0x202d
+      - 0x2105
       -
       -
       -
@@ -7630,7 +7630,7 @@ the following codes.
     * .. _MEDIA-BUS-FMT-UYYVYY16-4X32:
 
       - MEDIA_BUS_FMT_UYYVYY16_4X32
-      - 0x202f
+      - 0x2106
       -
       - u\ :sub:`15`
       - u\ :sub:`14`
@@ -7772,7 +7772,7 @@ the following codes.
     * .. _MEDIA-BUS-FMT-UYVY16-2X32:
 
       - MEDIA_BUS_FMT_UYVY16_2X32
-      - 0x2030
+      - 0x2108
       -
       - u\ :sub:`15`
       - u\ :sub:`14`
@@ -8032,7 +8032,7 @@ The following table list existing packed 36bit wide YUV formats.
     * .. _MEDIA-BUS-FMT-VUY12-1X36:
 
       - MEDIA_BUS_FMT_VUY12_1X36
-      - 0x2033
+      - 0x2104
       -
       - v\ :sub:`11`
       - v\ :sub:`10`
@@ -8243,7 +8243,7 @@ The following table list existing packed 48bit wide YUV formats.
     * .. _MEDIA-BUS-FMT-VUY16-1X48:
 
       - MEDIA_BUS_FMT_VUY16_1X48
-      - 0x2034
+      - 0x2107
       -
       -
       -
diff --git a/include/uapi/linux/media-bus-format.h b/include/uapi/linux/media-bus-format.h
index ef24f28d3ee1..b63b8d03b1a2 100644
--- a/include/uapi/linux/media-bus-format.h
+++ b/include/uapi/linux/media-bus-format.h
@@ -34,7 +34,7 @@
 
 #define MEDIA_BUS_FMT_FIXED			0x0001
 
-/* RGB - next is	0x1020 */
+/* RGB - next is	0x101d */
 #define MEDIA_BUS_FMT_RGB444_1X12		0x1016
 #define MEDIA_BUS_FMT_RGB444_2X8_PADHI_BE	0x1001
 #define MEDIA_BUS_FMT_RGB444_2X8_PADHI_LE	0x1002
@@ -63,11 +63,8 @@
 #define MEDIA_BUS_FMT_RGB101010_1X30		0x1018
 #define MEDIA_BUS_FMT_RGB121212_1X36		0x1019
 #define MEDIA_BUS_FMT_RGB161616_1X48		0x101a
-#define MEDIA_BUS_FMT_RBG101010_1X30		0x101d
-#define MEDIA_BUS_FMT_RBG121212_1X36		0x101e
-#define MEDIA_BUS_FMT_RBG161616_1X48		0x101f
 
-/* YUV (including grey) - next is	0x2035 */
+/* YUV (including grey) - next is	0x202d */
 #define MEDIA_BUS_FMT_Y8_1X8			0x2001
 #define MEDIA_BUS_FMT_UV8_1X8			0x2015
 #define MEDIA_BUS_FMT_UYVY8_1_5X8		0x2002
@@ -84,13 +81,11 @@
 #define MEDIA_BUS_FMT_VYUY10_2X10		0x2019
 #define MEDIA_BUS_FMT_YUYV10_2X10		0x200b
 #define MEDIA_BUS_FMT_YVYU10_2X10		0x200c
-#define MEDIA_BUS_FMT_VYYUYY10_4X20		0x2031
 #define MEDIA_BUS_FMT_Y12_1X12			0x2013
 #define MEDIA_BUS_FMT_UYVY12_2X12		0x201c
 #define MEDIA_BUS_FMT_VYUY12_2X12		0x201d
 #define MEDIA_BUS_FMT_YUYV12_2X12		0x201e
 #define MEDIA_BUS_FMT_YVYU12_2X12		0x201f
-#define MEDIA_BUS_FMT_VUY12_1X36		0x2033
 #define MEDIA_BUS_FMT_UYVY8_1X16		0x200f
 #define MEDIA_BUS_FMT_VYUY8_1X16		0x2010
 #define MEDIA_BUS_FMT_YUYV8_1X16		0x2011
@@ -108,19 +103,12 @@
 #define MEDIA_BUS_FMT_YUYV12_1X24		0x2022
 #define MEDIA_BUS_FMT_YVYU12_1X24		0x2023
 #define MEDIA_BUS_FMT_YUV10_1X30		0x2016
-#define MEDIA_BUS_FMT_VUY10_1X30		0x2032
 #define MEDIA_BUS_FMT_UYYVYY10_0_5X30		0x2027
 #define MEDIA_BUS_FMT_AYUV8_1X32		0x2017
 #define MEDIA_BUS_FMT_UYYVYY12_0_5X36		0x2028
 #define MEDIA_BUS_FMT_YUV12_1X36		0x2029
 #define MEDIA_BUS_FMT_YUV16_1X48		0x202a
-#define MEDIA_BUS_FMT_VUY16_1X48		0x2034
 #define MEDIA_BUS_FMT_UYYVYY16_0_5X48		0x202b
-#define MEDIA_BUS_FMT_VYYUYY8_1X24		0x202c
-#define MEDIA_BUS_FMT_Y16_1X16			0x202d
-#define MEDIA_BUS_FMT_UYYVYY12_4X24		0x202e
-#define MEDIA_BUS_FMT_UYYVYY16_4X32		0x202f
-#define MEDIA_BUS_FMT_UYVY16_2X32		0x2030
 
 /* Bayer - next is	0x3021 */
 #define MEDIA_BUS_FMT_SBGGR8_1X8		0x3001
@@ -167,4 +155,20 @@
 /* HSV - next is	0x6002 */
 #define MEDIA_BUS_FMT_AHSV8888_1X32		0x6001
 
+/* RGB: Xilinx Specific - next is        0x1103 */
+#define MEDIA_BUS_FMT_RBG101010_1X30		0x1100
+#define MEDIA_BUS_FMT_RBG121212_1X36		0x1101
+#define MEDIA_BUS_FMT_RBG161616_1X48		0x1102
+
+/* YUV: Xilinx Specific - next is       0x2109 */
+#define MEDIA_BUS_FMT_VYYUYY8_1X24		0x2100
+#define MEDIA_BUS_FMT_VYYUYY10_4X20		0x2101
+#define MEDIA_BUS_FMT_VUY10_1X30		0x2102
+#define MEDIA_BUS_FMT_UYYVYY12_4X24		0x2103
+#define MEDIA_BUS_FMT_VUY12_1X36		0x2104
+#define MEDIA_BUS_FMT_Y16_1X16			0x2105
+#define MEDIA_BUS_FMT_UYYVYY16_4X32		0x2106
+#define MEDIA_BUS_FMT_VUY16_1X48		0x2107
+#define MEDIA_BUS_FMT_UYVY16_2X32		0x2108
+
 #endif /* __LINUX_MEDIA_BUS_FORMAT_H */
-- 
2.31.1

